pkg/nats/transport/nats.go

```diff
@@ -48,6 +50,17 @@ type NATSTransportConfig struct {
 	ClusterPort              int
 	ClusterAdvertisedAddress string
 	ClusterPeers             []string
+
+	// TLS
+	ServerTLSCert    string
+	ServerTLSKey     string
+	ServerTLSTimeout int
+
+	// Used by the Nats Client when node acts as orchestrator
+	ServerTLSCACert string
+
+	// Use by the Nats Client when node acts as compute
+	ClientTLSCACert string
```

Comment:
what if you are running a hybrid node? do you think this differentiating is helpful here?

Reply:
I intentionally separated these 2 to make it explicit separation, with configuration coming from 2 different places: Orchestrator and Compute.  For hybrid, I would keep it as is, even if we defined it is 2 places, since technically we have a "compute node" and an "orchestrator node" concept.